src/transformers/models/flava/processing_flava.py

```diff
@@ -0,0 +1,146 @@
+# coding=utf-8
+# Copyright 2021 Meta Platforms authors and The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Image/Text processor class for FLAVA
+"""
+from typing import List, Optional, Union
+
+import numpy as np
+
+from transformers.data.data_collator import DataCollatorForWholeWordMask, tolist
+
+from ...processing_utils import ProcessorMixin
+from ...tokenization_utils_base import BatchEncoding, PaddingStrategy, PreTokenizedInput, TextInput, TruncationStrategy
+from ...utils import TensorType
+
+
+class FLAVAProcessor(ProcessorMixin):
+    r"""
+    Constructs a FLAVA processor which wraps a FLAVA feature extractor and a FLAVA tokenizer into a single processor.
+
+    [`FLAVAProcessor`] offers all the functionalities of [`FLAVAFeatureExtractor`] and [`FLAVATokenizerFast`]. See the
+    [`~FLAVAProcessor.__call__`] and [`~FLAVAProcessor.decode`] for more information.
+
+    Args:
+        feature_extractor ([`FLAVAFeatureExtractor`]):
+            The feature extractor is a required input.
+        tokenizer ([`FLAVATokenizerFast`]):
+            The tokenizer is a required input.
+    """
+    feature_extractor_class = "FLAVAFeatureExtractor"
+    tokenizer_class = ("BertTokenizer", "BertTokenizerFast")
+
+    def __init__(self, feature_extractor, tokenizer, mlm_probability=0.15):
+        super().__init__(feature_extractor, tokenizer)
+        self.current_processor = self.feature_extractor
+        self.text_masker = DataCollatorForWholeWordMask(tokenizer, mlm=True, mlm_probability=mlm_probability)
+
+    def __call__(
+        self,
+        images: Optional[
+            Union[
+                "Image.Image",  # noqa
+                np.ndarray,
+                "torch.Tensor",  # noqa
+                List["Image.Image"],  # noqa
+                List[np.ndarray],
+                List["torch.Tensor"],  # noqa
```

Comment:
importing the objects behing an `if TYPE_CHECKING` would avoid those `# noqa`

Reply:
Decided to leave it as it is if that is okay.

Reply:
I hate those kinds of type annotation anyway (not sure who it's actually helping honestly, when it's long like this), so do as you prefer :-)